File: Java/BubbleSort.java
import java.util.ArrayList;
import java.util.Scanner;

public class BubbleSort {
    public static void main(String args[]) {
        ArrayList<Integer> arr = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter size of the list: ");
        int size = scanner.nextInt();
        System.out.println("Enter the elements: ");
        for (int i = 0; i < size; i++) {
            int el = scanner.nextInt();
            arr.add(el);
        }
        bubbleSort(arr);
        System.out.println("Sorted List: ");
        for (int num : arr)
            System.out.print(num + " "); 
        scanner.close();
    }

    public static void bubbleSort(ArrayList<Integer> arr) {
        int size = arr.size();
        boolean swapped;
        for(int i = 0; i < size - 1; i++) {
            swapped = false;
            for(int j = 0; j < size - i - 1; j++) {
                if(arr.get(j) > arr.get(j + 1)) {
                    int temp = arr.get(j);
                    arr.set(j, arr.get(j + 1));
                    arr.set(j + 1, temp);
                    swapped = true;
                }
            }
            if(!swapped)
                break;
        }
    }
}

File: Javascript/BubbleSort.js
function BubbleSort(arr) {
    let swapped = false;
    for (let i = 0; i < arr.length; i++) {
        swapped = false;
        for (let j = 0; j < arr.length - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                swapped = true;
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
            }
        }
        if (!swapped)
            break;
    }
}

let arr = [5, 4, 3, 2, 1];
console.log("ARRAY BEFORE SORTING:");
console.log(arr);
BubbleSort(arr)
console.log("ARRAY AFTER SORTING:");
console.log(arr);
